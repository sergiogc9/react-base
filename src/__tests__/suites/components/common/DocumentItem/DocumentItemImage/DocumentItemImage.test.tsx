import React from "react";
import { mount } from "enzyme";
import DocumentItemImage, { OwnProps as ComponentProps } from '@src/components/common/Document/Item/DocumentItemImage/DocumentItemImage';
import GA from '@src/lib/googleAnalytics';

function getWrapperWithProps(props: ComponentProps) {
	return mount(<DocumentItemImage {...props} ></DocumentItemImage>);
}

describe("Document item images", () => {

	let wrapper: any;

	beforeEach(() => {
		wrapper = getWrapperWithProps({
			mainImage: "fake",
			mainLink: "http://discover.sprint.launchmetrics.com/r/d/2GT8E24urGs=_ckkDAAAAAAAAAAAAAAAAAA==",
			secondaryImage: "fake",
			secondaryLink: "http://discover.sprint.launchmetrics.com/r/d/2GT8E24urGs=_ckkDAAAAAAAAAAAAAAAAAA==?cover=1",
		});
	});

	it("Should render two images", () => {
		expect(wrapper.find('.document-item-image-link')).toHaveLength(2);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Should render one image", () => {
		wrapper.setProps({
			secondaryImage: "",
			secondaryLink: ""
		});
		expect(wrapper.find('.document-item-image-link')).toHaveLength(1);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Should not render any image", () => {
		wrapper.setProps({
			mainImage: "",
			mainLink: "",
			secondaryImage: "",
			secondaryLink: ""
		});
		wrapper.find('img').simulate('error');
		expect(wrapper.contains('img')).toBe(false);
	});

	it("Should change active image", () => {
		wrapper.find(".document-item-image-link[type='secondary']").simulate('click');
		expect(wrapper.state("activeImage")).toBe('secondary');
	});

	it("Should not change active image", () => {
		const prevActiveImage = wrapper.state("activeImage");
		wrapper.find(".document-item-image-link.active-image").simulate('click');
		expect(wrapper.state("activeImage")).toBe(prevActiveImage);
	});

	it("Test image tracker", () => {
		GA.trackEvent = jest.fn();
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find(".document-item-image-link.active-image").simulate('click');
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
	});
});
