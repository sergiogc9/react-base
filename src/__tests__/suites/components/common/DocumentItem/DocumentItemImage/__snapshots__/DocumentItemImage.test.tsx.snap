// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document item images Should render one image 1`] = `"<div class=\\"document-item-image\\"><a href=\\"http://discover.sprint.launchmetrics.com/r/d/2GT8E24urGs=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" type=\\"main\\" class=\\"document-item-image-link active-image\\"><img src=\\"fake\\" alt=\\"\\"></a></div>"`;

exports[`Document item images Should render two images 1`] = `"<div class=\\"document-item-image two-images\\"><a href=\\"http://discover.sprint.launchmetrics.com/r/d/2GT8E24urGs=_ckkDAAAAAAAAAAAAAAAAAA==?cover=1\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" type=\\"secondary\\" class=\\"document-item-image-link\\"><img src=\\"fake\\" alt=\\"\\"></a><a href=\\"http://discover.sprint.launchmetrics.com/r/d/2GT8E24urGs=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" type=\\"main\\" class=\\"document-item-image-link active-image\\"><img src=\\"fake\\" alt=\\"\\"></a></div>"`;
